src/lib/utils.ts

```diff
@@ -4,7 +4,14 @@ import { EVENT_MAP } from './constants';
 
 const defaultHanlder = () => null;
 
-export const buildQueryString = (config: Record<string, string>): string => Object.keys(config).map((key) => `${key}=${config[key]}`).join('&');
+export const buildQueryString = (
+  config: Record<string, string | Record<string, string>>,
+): string => Object.keys(config).map((key) => {
+  if (typeof config[key] === 'string') {
+    return `${key}=${config[key]}`
+  }
+  return `${key}=${JSON.stringify(config[key])}`
```

Comment:
I believe this is not the "URLize" a nested object. On our own frontends, we do something like this:

```ts
const options = {
  some: 'value',
  other: {
    nested: 'value2',
    nested2: 'value3',
  },
};

const qs = 'some=value&other[nested]=value2&other[nested2]=value3';
```

Reply:
(Our frontend filters have some arrays as values and maybe nested values, might be worth looking there?)